### PR TITLE
[FIXED] Allow any type of ack to suppress auto-cleanup of consumer.

### LINKED ISSUE #3630: Inactive threshold for pull consumer treat message signals (ack/nak etc.) as interest.
### Proposed Change:

For pull consumers, consider message signals (+ACK/-NAK/+WPI/+TERM/+NXT) as activity/interest and reset the inactive threshold timer.

### PR DESCRIPTION
**Pull request by derekcollison**
Also modified some of the auto-cleanup logic for pull consumers.
 
Signed-off-by: Derek Collison <derek@nats.io>

Resolves #3630 

/cc @nats-io/core

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/consumer.go [hunk 1]
```diff
@@ -853,10 +853,17 @@ func (o *consumer) updateInactiveThreshold(cfg *ConsumerConfig) {
 	// Ephemerals will always have inactive thresholds.
 	if !o.isDurable() && cfg.InactiveThreshold <= 0 {
 		// Add in 1 sec of jitter above and beyond the default of 5s.
-		o.dthresh = JsDeleteWaitTimeDefault + time.Duration(rand.Int63n(1000))*time.Millisecond
+		o.dthresh = cfg.InactiveThreshold + 100*time.Millisecond + time.Duration(rand.Int63n(900))*time.Millisecond
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/consumer.go:856]: Agree.

File: server/consumer.go [hunk 2]
```diff
@@ -1286,12 +1293,25 @@ func (o *consumer) deleteNotActive() {
 			return
 		}
 	} else {
-		// These need to keep firing so reset first.
-		if o.dtmr != nil {
-			o.dtmr.Reset(o.dthresh)
+		// Pull mode.
+		elapsed := time.Since(o.waiting.last)
+		if elapsed <= o.cfg.InactiveThreshold {
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/consumer.go:1298]: We should make sure Inactive is set properly. With fix above I think we should be good.

[the remark this answers is not in the document]
Comment (kozlovic) [reply server/consumer.go:1298]: What fix? You know set o.dthresh properly, but here you are still using o.cfg.InactiveThreshold, which I believe is wrong.

**Review by kozlovic — CHANGES_REQUESTED** [review]

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**